<launch>

	<!--<rosparam param="moveit_controller_manager">moveit_simple_controller_manager/MoveItSimpleControllerManager</rosparam>
	<param name="moveit_controller_manager" value="moveit_simple_controller_manager/MoveItSimpleControllerManager"></param>
	-->

	<!-- By default, we are not in debug mode -->
	<arg name="debug" default="false" />
	<arg name="rviz" default="true" />
	<arg name="kinect" default="false" />

	<!--<node name="edwin_hardware_node" pkg="edwin_moveit_config" type="edwin_hardware_node"
		ns="edwin" output="screen">
		<remap from="/edwin/arm_cmd" to="/arm_cmd"/>
		<remap from="/edwin/joint_states" to="joint_states"/>
	</node>-->

	<!-- Load the URDF, SRDF and other .yaml configuration files on the param server -->
	<include file="$(find edwin_moveit_config)/launch/planning_context.launch">
		<arg name="load_robot_description" value="true"/>
	</include>

	<!-- If needed, broadcast static tf for robot root -->
	<node pkg="tf" type="static_transform_publisher" name="virtual_joint_broadcaster_0" args="0 0 0 0 0 0 odom base_link 100" />

	<!-- We do not have a robot connected, so publish fake joint states -->
	<!--<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
	<param name="/use_gui" value="false"/>
	<rosparam param="/source_list">[/move_group/fake_controller_joint_states]</rosparam>
  </node>-->

  <!--<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
	  <param name="/use_gui" value="false"/>
  </node>-->

  <!-- Given the published joint states, publish tf for the robot links -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true" output="screen" >
	  <remap from="joint_states" to="/edwin/joint_states"/> 
  </node>

  <!-- controller -->
  <rosparam command="load" file="$(find edwin_moveit_config)/config/control.yaml"/>
  <!-- load the controllers -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" output="screen" ns="/edwin" args="edwin_controller joint_state_controller"/>

  <!-- Run the main MoveIt executable without trajectory execution (we do not have controllers configured by default) -->
  <include file="$(find edwin_moveit_config)/launch/move_group.launch">
	  <arg name="allow_trajectory_execution" value="true"/>
	  <arg name="fake_execution" value="false"/>
	  <arg name="info" value="true"/>
	  <arg name="debug" value="$(arg debug)"/>
  </include>

  <!-- Run Rviz and load the default config to see the state of the move_group node -->
  <group if="$(arg rviz)">
	  <include file="$(find edwin_moveit_config)/launch/moveit_rviz.launch">
		  <arg name="kinect" value="$(arg kinect)"/>
		  <arg name="debug" value="$(arg debug)"/>
	  </include>
  </group>

</launch>
